lib/utils/kubernetes.go

```diff
@@ -219,6 +220,33 @@ func IsKubernetesLabel(key string) bool {
 	return false
 }
 
+// IsHeadlessService return true if the given service is a headless service
+// (explicitly without a cluster IP)
+func IsHeadlessService(service v1.Service) bool {
+	const headlessServiceClusterIP = "None"
+	return service.Spec.ClusterIP == headlessServiceClusterIP
+}
+
+// IsAPIServerService return true if the given service specifies the API server service
+func IsAPIServerService(service v1.Service) bool {
+	const apiServerService = "kubernetes"
+	return service.Name == apiServerService && service.Namespace == metav1.NamespaceDefault
+}
+
+// WithService returns a new logger with service-relevant metadata
+func WithService(service v1.Service, logger log.FieldLogger) log.FieldLogger {
```

Comment:
Nit: I'd consider renaming to `LoggerWithService` or `LogWithService`, otherwise `utils.WithService()` reads a bit weird.